update.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+set -eo pipefail
+
+declare -A php_version=(
+	[default]='7.1'
+)
+
+declare -A cmd=(
+	[apache]='apache2-foreground'
+	[fpm]='php-fpm'
+	[fpm-alpine]='php-fpm'
+)
+
+declare -A base=(
+	[apache]='debian'
+	[fpm]='debian'
+	[fpm-alpine]='alpine'
+)
+
+declare -A extras=(
+	[apache]='\nRUN a2enmod rewrite remoteip ;\\\n    {\\\n     echo RemoteIPHeader X-Real-IP ;\\\n     echo RemoteIPTrustedProxy 10.0.0.0/8 ;\\\n     echo RemoteIPTrustedProxy 172.16.0.0/12 ;\\\n     echo RemoteIPTrustedProxy 192.168.0.0/16 ;\\\n    } > /etc/apache2/conf-available/remoteip.conf;\\\n    a2enconf remoteip'
+	[fpm]=''
+	[fpm-alpine]=''
+)
+
+declare -A pecl_versions=(
+	[Imagick]='3.4.3'
+)
+
+declare -A install_extras=(
+    ['stable']='\nRUN set -ex; \\\n    curl -fsSL -o friendica.tar.gz \\\n        "https://github.com/friendica/friendica/releases/download/${FRIENDICA_VERSION}/friendica-full-${FRIENDICA_VERSION}.tar.gz"; \\\n    tar -xzf friendica.tar.gz -C /usr/src/; \\\n    rm friendica.tar.gz; \\\n    mv -f /usr/src/friendica-${FRIENDICA_VERSION}/ /usr/src/friendica; \\\n    chmod 777 /usr/src/friendica/view/smarty3; \\\n    curl -fsSL -o friendica_addons.tar.gz \\\n        "https://github.com/friendica/friendica-addons/archive/${ADDONS_VERSION}.tar.gz"; \\\n    mkdir /usr/src/friendica/addon; \\\n    tar -xzf friendica_addons.tar.gz -C /usr/src/friendica/addon --strip-components=1; \\\n    rm friendica_addons.tar.gz;'
+    ['develop']=''
+)
+
+declare -A bin_dir=(
+  ['stable']='scripts'
+  ['develop']='bin'
+)
+
+variants=(
+	apache
+	fpm
+	fpm-alpine
+)
+
+versions=(
+    2018.05-rc
+)
+
+travisEnv=
+travisEnvAmd64=
+
+function create_variant() {
+	dir="$1/$variant"
+
+	# Create the version+variant directory with a Dockerfile.
+	mkdir -p "$dir"
+
+        template="Dockerfile-${base[$variant]}.template"
+        echo "# DO NOT EDIT: created by update.sh from $template" > "$dir/Dockerfile"
+	cat "$template" >> "$dir/Dockerfile"
+
+    # Check which installation typ we need. If develop, the source will get downloaded by git.
```

Comment:
Why only with develop?

Reply:
I assume that with stable the code will be embedded in the image itself and the image will be updated once each release. Since `develop` changes more frequently, the code has to be retrieved dynamically from git.

Reply:
@MrPetovan you're right. That's exactly the reason for it :-)

Reply:
thanks for the clarification :-)